### PR TITLE
Bugfix/iab dashboard group count and more

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: reporting-service/src/main/resources/application-no-permission.sql.yml [hunk 1]
```diff
@@ -39,11 +39,9 @@ sql:
       SELECT
         tsg.id,
         tsg.name,
-        s.code as subject_code,
-        count(tsgm.student_id) as student_count
```
Comment (esotrope) [inline reporting-service/src/main/resources/application-no-permission.sql.yml]: turns out we don't need this in the ui when doing a findAll on user groups.
yw @agorina

File: webapp/src/main/webapp/src/app/dashboard/group-dashboard/group-dashboard.component.ts [hunk 2]
```diff
@@ -41,17 +42,16 @@ export class GroupDashboardComponent implements OnInit {
   ngOnInit() {
     const { groupId, userGroupId, schoolYear } = this.route.snapshot.params;
     forkJoin(
+      groupId != null
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/dashboard/group-dashboard/group-dashboard.component.ts]: adds back the call to get a singular group so we have the student count in hand. ty for the catch @jtreuting

File: webapp/src/main/webapp/src/app/groups/group-table.component.html [hunk 3]
```diff
@@ -30,7 +30,7 @@
 
         <div *ngSwitchCase="'group'"
              class="label-max-width">
-          <a [routerLink]="['group-exams', {groupId: group.id}]"
+          <a [routerLink]="['/', 'group-exams', {groupId: group.id}]"
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/groups/group-table.component.html]: first attempt at resolving 1901. we'll see how it goes

File: reporting-service/src/main/resources/application-no-permission.sql.yml (hunk 1 above)
Comment (agorina) [reply reporting-service/src/main/resources/application-no-permission.sql.yml]: 👍

**Pull request merged**